### PR TITLE
Execute NetworkBenchmarks from Flink

### PR DESCRIPTION
**Pull request by pnowojski**

### CODE REVIEW

File: src/test/java/org/apache/flink/benchmark/StreamNetworkThroughputBenchmarkExecutor.java [hunk 1]
```diff
@@ -62,20 +63,37 @@ public void networkThroughput(MultiEnvironment context) throws Exception {
...
 		@Setup
 		public void setUp() throws Exception {
+			int channels = parseChannels(channelsFlushTimeout);
+			int flushTimeout = parseFlushTimeout(channelsFlushTimeout);
 			super.setUp(writers, channels, flushTimeout);
 		}
 
+		private static int parseFlushTimeout(String channelsFlushTimeout) {
+			String[] parameters = channelsFlushTimeout.split(",");
+			checkState(parameters.length >= 2);
```
Comment (NicoK) [inline src/test/java/org/apache/flink/benchmark/StreamNetworkThroughputBenchmarkExecutor.java]: shouldn't it rather be `checkArgument`?

File: src/test/java/org/apache/flink/benchmark/StreamNetworkThroughputBenchmarkExecutor.java [hunk 2]
```diff
@@ -62,20 +63,37 @@ public void networkThroughput(MultiEnvironment context) throws Exception {
...
-		@Param({"1", "100"})
-		public int flushTimeout = 1;
+		//Do not spam continuous benchmarking with number of writers parameter.
+		//@Param({"1", "4"})
+		public int writers = 4;
 
 		@Setup
 		public void setUp() throws Exception {
+			int channels = parseChannels(channelsFlushTimeout);
+			int flushTimeout = parseFlushTimeout(channelsFlushTimeout);
```
Comment (NicoK) [inline src/test/java/org/apache/flink/benchmark/StreamNetworkThroughputBenchmarkExecutor.java]: At some point, we should do the parsing only once, but imho, this is ok for now.

File: src/test/java/org/apache/flink/benchmark/StreamNetworkThroughputBenchmarkExecutor.java [hunk 3]
```diff
@@ -62,20 +63,37 @@ public void networkThroughput(MultiEnvironment context) throws Exception {
...
 			super.setUp(writers, channels, flushTimeout);
 		}
 
+		private static int parseFlushTimeout(String channelsFlushTimeout) {
+			String[] parameters = channelsFlushTimeout.split(",");
+			checkState(parameters.length >= 2);
+			String flushTimeout = parameters[1];
+
+			checkState(flushTimeout.endsWith("ms"));
```
Comment (NicoK) [inline src/test/java/org/apache/flink/benchmark/StreamNetworkThroughputBenchmarkExecutor.java:87]: `checkArgument`?

File: src/test/java/org/apache/flink/benchmark/StreamNetworkThroughputBenchmarkExecutor.java [hunk 4]
```diff
@@ -62,20 +63,37 @@ public void networkThroughput(MultiEnvironment context) throws Exception {
...
+			return Integer.parseInt(flushTimeout.substring(0, flushTimeout.length() - 2));
+		}
+
+		private static int parseChannels(String channelsFlushTimeout) {
+			String[] parameters = channelsFlushTimeout.split(",");
+			checkState(parameters.length >= 1);
```
Comment (NicoK) [inline src/test/java/org/apache/flink/benchmark/StreamNetworkThroughputBenchmarkExecutor.java:93]: `checkArgument`?

**Review by NicoK — CHANGES_REQUESTED** [review]
ok, I would just change from `checkState` to `checkArgument` - rest looks ok, basically

File: src/test/java/org/apache/flink/benchmark/StreamNetworkThroughputBenchmarkExecutor.java (hunk 2 above)
Comment (pnowojski) [reply src/test/java/org/apache/flink/benchmark/StreamNetworkThroughputBenchmarkExecutor.java]: Yes, agree, I'm not proud of it, but I didn't want to over-engineer it at this moment by introducing additional pojo to return both values at once :(

**Comment by pnowojski** [comment]
Thanks for the review! merged

**Pull request closed without merging**